Sources/SwiftParser/TokenPrecedence.swift

```diff
@@ -101,8 +101,8 @@ public enum TokenPrecedence: Comparable {
 
   @_spi(RawSyntax)
   public init(_ lexeme: Lexer.Lexeme) {
-    if lexeme.rawTokenKind == .keyword {
-      self.init(Keyword(lexeme.tokenText)!)
+    if let kw = Keyword(lexeme.tokenText) {
+      self.init(kw)
```

Comment:
This is not correct because it looses the distinction of e.g. a token that has text `open` e.g. in `self.open` and is an identifier and `open` that is considered a keyword e.g. `open func foo() {}`.

Also, performance-wise, iterating over all different keyword text choices is a lot more expensive than doing a scalar type comparison for the lexeme's `rawTokenKind`.